From 6e7ed5f8e2f6ab4ff8b83e01fe3072315eb17839 Mon Sep 17 00:00:00 2001
From: Ricardo Salveti <ricardo@foundries.io>
Date: Mon, 30 Jul 2018 15:27:57 -0300
Subject: [PATCH] ostreemanager: tag /run/aktualizr/ostree-pending-update after
 updates

Have an easy way for userspace to know when there is a pending ostree
update, allowing custom automatic reboot mechanisms.

Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
---
 src/libaktualizr/package_manager/ostreemanager.cc | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/libaktualizr/package_manager/ostreemanager.cc b/src/libaktualizr/package_manager/ostreemanager.cc
index 73514953..54de1945 100644
--- a/src/libaktualizr/package_manager/ostreemanager.cc
+++ b/src/libaktualizr/package_manager/ostreemanager.cc
@@ -141,6 +141,16 @@ data::InstallOutcome OstreeManager::install(const Uptane::Target &target) const
   }
   LOG_INFO << "Performing sync()";
   sync();
+
+  LOG_INFO << "Creating /run/aktualizr/ostree-pending-update";
+  boost::filesystem::create_directory("/run/aktualizr");
+  std::ofstream file("/run/aktualizr/ostree-pending-update");
+  if (!file.good()) {
+    LOG_ERROR << "Failed to open /run/aktualizr/ostree-pending-update";
+  }
+  file.close();
+  LOG_INFO << "Created empty /run/aktualizr/ostree-pending-update";
+
   return data::InstallOutcome(data::OK, "Installation successful");
 }
 
-- 
2.18.0

